include/revng/Yield/Support/PTML.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <climits>
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringMap.h"
+
+#include "revng/ADT/Concepts.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/YAMLTraits.h"
+
+namespace yield::ptml {
+
+namespace tags {
+
+inline constexpr auto Div = "div";
+inline constexpr auto Span = "span";
+
+} // namespace tags
+
+namespace attributes {
+
+inline constexpr auto scope = "data-scope";
+inline constexpr auto token = "data-token";
+inline constexpr auto locationDefinition = "data-location-definition";
+inline constexpr auto locationReferences = "data-location-references";
+inline constexpr auto modelEditPath = "data-model-edit-path";
+
+} // namespace attributes
+
+namespace scopes {
+
+inline constexpr auto Annotation = "annotation";
+inline constexpr auto Comment = "comment";
+inline constexpr auto Error = "error";
+inline constexpr auto Link = "link";
+
+} // namespace scopes
+
+namespace tokens {
+
+inline constexpr auto Indentation = "indentation";
+
+} // namespace tokens
+
+class Tag {
+private:
+  std::string TheTag;
+  std::string Content;
+  llvm::StringMap<std::string> Attributes;
+
+public:
+  Tag() {}
+  explicit Tag(llvm::StringRef Tag) : TheTag(Tag.str()) {}
+  explicit Tag(llvm::StringRef Tag, llvm::StringRef Content) :
+    TheTag(Tag.str()), Content(Content.str()) {}
+
+  Tag &setContent(const llvm::StringRef Content) {
+    this->Content = Content.str();
+    return *this;
+  }
+
+  Tag &add(llvm::StringRef Name, llvm::StringRef Value) {
+    Attributes[Name] = Value.str();
+    return *this;
+  }
+
+  template<range_with_value_type<llvm::StringRef> T>
+  Tag &add(llvm::StringRef Name, const T &Values) {
+    Attributes[Name] = llvm::join(Values, ",");
+    return *this;
+  }
+
+  template<typename... T>
+  requires(std::is_convertible_v<llvm::StringRef, T> and...)
+    Tag &add(llvm::StringRef Name, T &&...Values) {
+    Attributes[Name] = llvm::join(",", std::forward<T>(Values)...);
+    return *this;
+  }
```

Comment:
I'd argue on the contrary, there should be only one way to represent a list of items in PTML (e.g. values concatenated with `,`), otherwise it'll lead to either:
* Fragmentation if at any point we decide that some attributes are concatenated differently
* Code duplication because each caller will exactly do the same thing copypasted

As for a name, I'd go with `addAttributeList` or something along those lines

Reply:
If you put this here, it's an actual extension of the "PTML" format so:
1. Document it in the pad (which will eventually become doc)
2. Assert no `,` in the values
3. Go for `addListAttribute`

I'm not a huge fan of this, but if it happens often that we need list of comma-separated things, let's go for it.

Are you sure we don't want to use `' '` as a separator? Would be similar to the `<div class="...">` attribute.

Reply:
1. will do
2. done
3. ok

> Are you sure we don't want to use `'  '` as a separator?

I'd stick with `,` since a list of `Target`s is already concatenated with `,` and it would be nice to remain consistent throughout the codebase. (` ` is still fine if we want to use it)